connection.go

```diff
@@ -502,6 +522,25 @@ func (conn *Connection) dial() (err error) {
 	conn.Greeting.Version = bytes.NewBuffer(greeting[:64]).String()
 	conn.Greeting.auth = bytes.NewBuffer(greeting[64:108]).String()
 
+	// IPROTO_ID requests can be processed without authentication.
+	// https://www.tarantool.io/en/doc/latest/dev_guide/internals/iproto/requests/#iproto-id
+	if err = conn.identify(w, r); err != nil {
+		connection.Close()
+		return err
+	}
+
+	if err = checkProtocolVersion(opts.RequiredProtocolVersion, conn.ServerProtocolVersion()); err != nil {
+		connection.Close()
+		return fmt.Errorf("server: %w", err)
+	}
+
+	requiredFeatures := make([]Feature, len(opts.RequiredFeatures))
+	copy(requiredFeatures, opts.RequiredFeatures)
```

Comment:
dial() may be called much later of Connect() (on the reconnect case), see the problem:

https://github.com/tarantool/go-tarantool/pull/226#discussion_r1025243600

It is better to save a copy of the slice at the beginning of Connect() call.

Reply:
Added opts copy